packages/client/src/__tests__/integration/happy/client-engine/test.ts

```diff
@@ -0,0 +1,120 @@
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+import {
+  ClientEngineType,
+  DEFAULT_CLIENT_ENGINE_TYPE,
+} from '../../../../runtime/utils/getClientEngineType'
+import { generateTestClient } from '../../../../utils/getTestClient'
+
+const buildSchema = (engineType?: string) => `
+datasource db {
+  provider = "sqlite"
+  url      = "file:./dev.db"
+}
+
+generator client {
+  provider = "prisma-client-js"
+  ${engineType ? `engineType="${engineType}"` : ''}
+}
+
+// / User model comment
+model User {
+  id    String  @default(uuid()) @id
+  email String  @unique
+  // / name comment
+  name  String?
+  posts Post[]
+  profile Profile?
+}
+
+model Profile {
+  id     String     @default(cuid()) @id
+  bio    String?
+  user   User    @relation(fields: [userId], references: [id])
+  userId String     @unique
+}
+
+model Post {
+  id        String   @default(cuid()) @id
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+  published Boolean
+  title     String
+  content   String?
+  authorId  String? @map("author")
+  author    User?    @relation(fields: [authorId], references: [id])
+}
+`
+function getExpectedEngine(engineType, envVar, envVarValue) {
+  if (envVar === 'PRISMA_CLIENT_ENGINE_TYPE' && envVarValue) {
+    return envVarValue
+  }
+  if (engineType) {
+    return engineType
+  }
+
+  return DEFAULT_CLIENT_ENGINE_TYPE
+}
+function buildTests() {
+  const engineTypes = [
+    ClientEngineType.Binary,
+    ClientEngineType.NodeAPI,
+    undefined,
+  ]
+  const envVars = {
+    PRISMA_CLIENT_ENGINE_TYPE: engineTypes,
+  }
+  engineTypes.forEach((engineType) => {
+    for (const envVar in envVars) {
+      envVars[envVar].forEach((value) => {
```

Comment:
Would be nice to use for in everywhere I think? Would be a bit easier to read

Reply:
`for in` is mainly for objects changed them to `for of`